src/plugins/bfetch/public/streaming/fetch_streaming.ts

```diff
@@ -31,49 +30,39 @@ export function fetchStreaming({
   method = 'POST',
   body = '',
   signal,
-  compressionDisabled$ = of(false),
+  getIsCompressionDisabled = () => false,
 }: FetchStreamingParams) {
   const xhr = new window.XMLHttpRequest();
 
-  const msgStream = compressionDisabled$.pipe(
-    switchMap((compressionDisabled) => {
-      // Begin the request
-      xhr.open(method, url);
-      xhr.withCredentials = true;
+  // Begin the request
+  xhr.open(method, url);
+  xhr.withCredentials = true;
 
-      if (!compressionDisabled) {
-        headers['X-Chunk-Encoding'] = 'deflate';
-      }
+  const isCompressionDisabled = getIsCompressionDisabled();
 
-      // Set the HTTP headers
-      Object.entries(headers).forEach(([k, v]) => xhr.setRequestHeader(k, v));
+  if (!isCompressionDisabled) {
+    headers['X-Chunk-Encoding'] = 'deflate';
+  }
 
-      const stream = fromStreamingXhr(xhr, signal);
+  // Set the HTTP headers
+  Object.entries(headers).forEach(([k, v]) => xhr.setRequestHeader(k, v));
 
-      // Send the payload to the server
-      xhr.send(body);
+  const stream = fromStreamingXhr(xhr, signal);
 
-      // Return a stream of chunked decompressed messages
-      return stream.pipe(
-        split('\n'),
-        map((msg) => {
-          return compressionDisabled ? msg : inflateResponse(msg);
-        })
-      );
+  // Send the payload to the server
+  xhr.send(body);
+
+  // Return a stream of chunked decompressed messages
+  const stream$ = stream.pipe(
+    split('\n'),
+    map((msg) => {
+      return isCompressionDisabled ? msg : inflateResponse(msg);
     }),
     share()
   );
 
-  // start execution
-  const msgStreamSub = msgStream.subscribe({
```

Comment:
Memory leak came from this observable never unsubscribed because `msgStream` never complete because `compressionDisabled$` never completes. 

Most likely, one line fix could be: 
```
const msgStream = compressionDisabled$.pipe(
     take(1),
     switchMap((compressionDisabled)
```

But it seemed that we don't need compressionDisabled$ as observable here so I decided to simplify and remove it